# 🏪 Fase 2.2: Exploratory Data Analysis (EDA) - Dataset Stores

## Objetivo

El dataset **Stores** contiene la información descriptiva de cada tienda, incluyendo ciudad, estado, tipo y clúster.

El objetivo de este análisis es comprender la estructura de la red de tiendas e identificar características que puedan influir sobre el comportamiento de las ventas.

## Preguntas de Negocio

Durante este análisis se buscará responder las siguientes preguntas:

1. ¿Cuántas tiendas existen en el negocio?
2. ¿Cómo se distribuyen las tiendas por ciudad y estado?
3. ¿Qué tipos de tiendas existen?
4. ¿Cómo se distribuyen los clústeres de tiendas?
5. ¿Existen ciudades con mayor concentración de tiendas?
6. ¿Las características de una tienda podrían explicar diferencias en las ventas?
7. ¿Existen posibles inconsistencias o valores faltantes en la información de las tiendas?

In [1]:
# carga de librerias

import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

# configuracion de pandas de mostrar todas las columnas
pd.set_option("display.max_columns", None)

# configuracion de pandas decimales a mostrar
pd.set_option("display.float_format", "{:,.2f}".format)

# direccion de los datos
PATH = "../data/raw/"

In [2]:
# Carga de datos

stores = pd.read_csv(PATH + "stores.csv")

## Creación del data_clean

Para preservar la integridad del conjunto de datos original, se crea una copia denominada `stores_clean`. A partir de este punto, todas las tareas de limpieza, transformación y análisis exploratorio se realizarán sobre esta copia.

**Objetivo:** mantener el dataset `stores` sin modificaciones para poder utilizarlo como referencia durante todo el proyecto.

In [3]:
# Creacion de copia de seguridad de los datos
stores_clean = stores.copy()

In [4]:
# Exploracion de los datos
stores_clean.info()

<class 'pandas.DataFrame'>
RangeIndex: 54 entries, 0 to 53
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype
---  ------     --------------  -----
 0   store_nbr  54 non-null     int64
 1   city       54 non-null     str  
 2   state      54 non-null     str  
 3   type       54 non-null     str  
 4   cluster    54 non-null     int64
dtypes: int64(2), str(3)
memory usage: 2.2 KB


## Diccionario de Variables

| Columna | Descripción | Tipo de dato |
|----------|-------------|--------------|
| store_nbr | Identificador único de la tienda | `int64` |
| city | Ciudad donde se encuentra ubicada la tienda | `str` |
| state | Estado o provincia donde se ubica la tienda | `str` |
| type | Tipo o categoría de la tienda | `str` |
| cluster | Identificador del clúster al que pertenece la tienda | `int64` |

### Observaciones

- **store_nbr** es el identificador único de cada tienda y será utilizado para relacionar este dataset con el historial de ventas (`train`).
- **city** y **state** son variables categóricas que permiten realizar análisis geográficos y regionales.
- **type** clasifica las tiendas según su categoría, lo que permitirá evaluar posibles diferencias en el comportamiento de las ventas.
- **cluster** representa una agrupación de tiendas con características similares. Aunque está almacenado como `int64`, corresponde a una variable categórica y no debe interpretarse como un valor numérico continuo.

- El dataset no presenta valores nulos, por lo que inicialmente no se requieren procesos de limpieza relacionados con datos faltantes.

In [5]:
# Exploracion de los datos
stores_clean.head()

,store_nbr,city,state,type,cluster
0,1,Quito,Pichincha,D,13
1,2,Quito,Pichincha,D,13
2,3,Quito,Pichincha,D,8
3,4,Quito,Pichincha,D,9
4,5,Santo Domingo,Santo Domingo de los Tsachilas,D,4


## Validación de variables numéricas

Como parte del proceso de calidad de datos, se revisan las variables numéricas del dataset **Stores**, verificando sus valores mínimos, máximos y la consistencia de los identificadores.

El objetivo es confirmar que la información descriptiva de las tiendas sea coherente y detectar posibles anomalías, tales como:

- Identificadores de tiendas fuera del rango esperado.
- Clústeres con valores inválidos o inconsistentes.


Esta revisión permite garantizar la integridad del dataset antes de utilizarlo en el análisis exploratorio y posteriormente integrarlo con el historial de ventas.

In [6]:
# Exploracion de los datos
stores_clean.describe()

,store_nbr,cluster
count,54.00,54.00
mean,27.50,8.48
std,15.73,4.69
min,1.00,1.00
25%,14.25,4.00
50%,27.50,8.50
75%,40.75,13.00
max,54.00,17.00


## Análisis de las variables numéricas

A continuación se presentan las principales observaciones obtenidas a partir de las estadísticas descriptivas de las variables numéricas del dataset **Stores**.

### Observaciones

- **store_nbr**
  - Existen **54 tiendas**, identificadas con valores consecutivos entre **1** y **54**.
  - Cada identificador corresponde a una tienda única.
  - Aunque está almacenada como un número entero (`int64`), representa una **variable categórica** y no una magnitud numérica.

- **cluster**
  - Los clústeres toman valores entre **1** y **17**, indicando que las tiendas han sido agrupadas en **17 segmentos** según características comunes definidas por el negocio.
  - La mediana es **8.5**, mientras que el promedio es **8.48**, lo que sugiere una distribución relativamente equilibrada de los identificadores de los clústeres.
  - Al igual que `store_nbr`, esta variable está codificada numéricamente, pero representa una **categoría** y no un valor continuo.
  - Las estadísticas descriptivas (media, desviación estándar y cuartiles) únicamente describen la distribución de los identificadores y **no deben interpretarse como una medida cuantitativa del negocio**.

### Conclusiones

- Las variables **`store_nbr`** y **`cluster`** deben tratarse como **variables categóricas** durante el análisis exploratorio y el modelado.

- No se observan valores fuera de los rangos esperados para los identificadores de tiendas ni para los clústeres.

**El dataset presenta una estructura consistente para las variables numéricas, por lo que no se evidencian anomalías que requieran acciones de limpieza en esta etapa.**

## Respondiendo Preguntas de Negocio

Durante este análisis se buscará responder las siguientes preguntas:

1. ¿Cuántas tiendas existen en el negocio?
2. ¿Cómo se distribuyen las tiendas por ciudad y estado?
3. ¿Qué tipos de tiendas existen?
4. ¿Cómo se distribuyen los clústeres de tiendas?
5. ¿Existen ciudades con mayor concentración de tiendas?
6. ¿Las características de una tienda podrían explicar diferencias en las ventas?
7. ¿Existen posibles inconsistencias o valores faltantes en la información de las tiendas?